### PR TITLE
Only listen to the fullscreen-events of one of the different vendor

### PR DESCRIPTION
**Pull request by Abszissex**
Without doing this, it can happen that the handlers are called via the 'fullscreenchange' and 'webkitfullscreenchange' event when using the 'Fullscreen-API-Polyfill' (https://github.com/neovov/Fullscreen-API-Polyfill). So it is best to only listen to one of the events, otherwise the handler may be called several times which lead to nothing being presented.

### CODE REVIEW

File: src/base.js [hunk 1]
```diff
@@ -330,17 +330,17 @@ VRDisplay.prototype.addFullscreenListeners_ = function(element, changeHandler, e
   this.fullscreenErrorHandler_ = errorHandler;
 
   if (changeHandler) {
-    element.addEventListener('fullscreenchange', changeHandler, false);
-    element.addEventListener('webkitfullscreenchange', changeHandler, false);
-    document.addEventListener('mozfullscreenchange', changeHandler, false);
-    element.addEventListener('msfullscreenchange', changeHandler, false);
+    if (document.fullscreenEnabled) element.addEventListener('fullscreenchange', changeHandler, false);
+	  else if (document.webkitFullscreenEnabled) element.addEventListener('webkitfullscreenchange', changeHandler, false);
```
Comment (borismus) [inline src/base.js]: Please use braces `{`, `}`, and ensure that spacing is consistent with the rest of the code.

File: src/base.js [hunk 2]
```diff
@@ -330,17 +330,17 @@ VRDisplay.prototype.addFullscreenListeners_ = function(element, changeHandler, e
...
+	  else if (document.mozFullscreenEnabled) document.addEventListener('mozfullscreenchange', changeHandler, false);
```
Comment (cvan) [inline src/base.js]: typo here: should be `document.mozFullScreenEnabled` (notice the capitalised `S` in `Screen`)

File: src/base.js [hunk 3]
```diff
@@ -330,17 +330,17 @@ VRDisplay.prototype.addFullscreenListeners_ = function(element, changeHandler, e
...
   }
 
   if (errorHandler) {
-    element.addEventListener('fullscreenerror', errorHandler, false);
-    element.addEventListener('webkitfullscreenerror', errorHandler, false);
-    document.addEventListener('mozfullscreenerror', errorHandler, false);
-    element.addEventListener('msfullscreenerror', errorHandler, false);
+    if (document.fullscreenEnabled) element.addEventListener('fullscreenerror', errorHandler, false);
+	  else if (document.webkitFullscreenEnabled) element.addEventListener('webkitfullscreenerror', errorHandler, false);
+	  else if (document.mozFullscreenEnabled) document.addEventListener('mozfullscreenerror', errorHandler, false);
```
Comment (cvan) [inline src/base.js]: same here

**Comment by Abszissex** [comment]
Thanks for your replies guys.
Sorry 'bout the braces and the mozFullScreenEnabled. Done this after 10 hours of work without break. The next time I will take more care about writing more cleaner code for PR.
Applied the changes u mentioned to my fork. You should be able to merge it in now, shouldn't it? 
Sorry it is more or less my first PR :/

**Comment by cvan** [comment]
@Abszissex no worries, this looks great - nice work! I don't have merge rights, so will have to wait for @borismus. thanks 👍

**Comment by Abszissex** [comment]
Also I added the functionality to add a callback when triggering the back button in the cardboard UI and to add several viewers if u want the user to choose between more (not everyone have a cardboard but instead for example a Bobo VR or similar). 
unfortunately pushed them all on my master instead of creating a branch for each feature :/
what do you think about these two features?

File: src/cardboard-vr-display.js [hunk 4]
```diff
@@ -149,6 +149,11 @@ CardboardVRDisplay.prototype.beginPresent_ = function() {
       e.preventDefault();
     }.bind(this), function(e) {
       // Back clicked.
+
+      // In case of a custom calllback defined by the user, trigger it
+      if (typeof WebVRConfig.BACKACTION_CALLBACK === 'function') {
+        WebVRConfig.BACKACTION_CALLBACK();
```
Comment (cvan) [inline src/cardboard-vr-display.js]: should this be called before or after `exitPresent`. personally, as a dev, I'd want to call it after the `vrDisplay.exitPresent()` `Promise` resolves or rejects (or get a `Promise` back).

Comment (Abszissex) [inline src/cardboard-vr-display.js]: Yeah I see your point. 
I indeed agree with you that the callback should be made after the` exitPresent`

**Comment by PaHae** [comment]
@cvan  BACKACTION_CALLBACK is now called after the `exitPresent` is resolved/rejected

@borismus what do you think about the changes? Mind merging these?

PS: I am Abszissex

File: README.md [hunk 5]
```diff
@@ -87,7 +87,31 @@ WebVRConfig = {
   // Dirty bindings include: gl.FRAMEBUFFER_BINDING, gl.CURRENT_PROGRAM,
   // gl.ARRAY_BUFFER_BINDING, gl.ELEMENT_ARRAY_BUFFER_BINDING,
   // and gl.TEXTURE_BINDING_2D for texture unit 0.
-  DIRTY_SUBMIT_FRAME_BINDINGS: true // Default: false.
+  DIRTY_SUBMIT_FRAME_BINDINGS: true, // Default: false.
+
+  // A custom callback which will be called when the User triggers the back
+  // button in the CARDBOARD_UI
+  BACKACTION_CALLBACK: function() {
```
Comment (borismus) [inline README.md]: These changes don't belong here. Please remove.

File: build/webvr-polyfill.js [hunk 6]
```diff
@@ -1595,7 +1605,14 @@ CardboardVRDisplay.prototype.beginPresent_ = function() {
       e.preventDefault();
     }.bind(this), function(e) {
       // Back clicked.
-      this.exitPresent();
+
+      this.exitPresent().then(function () {
```
Comment (borismus) [inline build/webvr-polyfill.js]: Also, please remove.

File: build/webvr-polyfill.js [hunk 7]
```diff
@@ -1682,6 +1699,12 @@ CardboardVRDisplay.prototype.onResize_ = function(e) {
     // hide the URL bar unless content is bigger than the screen.
     // This will not be visible as long as the container element (e.g. body)
     // is set to 'overflow: hidden'.
+    var padding;
```
Comment (borismus) [inline build/webvr-polyfill.js]: Please remove, this is going to be coming in from another PR.

File: build/webvr-polyfill.js [hunk 8]
```diff
@@ -1973,6 +1996,13 @@ var DEFAULT_RIGHT_CENTER = {x: 0.5, y: 0.5};
  * params were found.
  */
 function DeviceInfo(deviceParams) {
+  if (WebVRConfig.ADDITIONAL_VIEWERS instanceof Array) {
```
Comment (borismus) [inline build/webvr-polyfill.js]: Please remove

**Comment by cvan** [comment]
when you get a chance - @Abszissex, could you address @borismus' comments?

**Comment by PaHae** [comment]
@cvan see #157 

@borismus thanks for you feedback, close this PR please and refer to #157 made a "clean" PR for the fullscreen-events
